eclipse-collections/src/main/java/org/eclipse/collections/impl/map/ordered/mutable/UnmodifiableMutableOrderedMap.java

```diff
@@ -1566,12 +1565,6 @@ public void forEachWithIndex(
         this.delegate.forEachWithIndex(fromIndex, toIndex, objectIntProcedure);
     }
 
-    @Override
```

Comment:
Why remove this implementation?

Reply:
This is a philosophical thing where we should just decide/agree what we're going to do. Real mutations are always concrete. There are some composite methods that mutate, like MutableMapIterable.putPair() but they always delegate to another concrete method like put(). So it's always safe to delete methods from Unmodifiable wrappers; I've yet to see any counter-example. On the other hand, it would be super-consistent to delegate all read-only methods to the delegate. But we don't do that either. UnmodifiableMutableOrderedMap is missing overrides of all sorts of things.

* Our own mutating methods, like putPair
* JDK mutating methods, like replaceAll
* Our own read-only methods, like countByEach
* JDK read-only methods, like forEach(BiConsumer)

There are something like ~20 methods that you might expect to be overridden and are not. Given the current state of affairs, I think I've made things more consistent with this change, even if we choose to go in a different direction later.

Reply:
Thanks for the clarification. I'm good with this.